src/math/deconvolution_2d.jl

```diff
@@ -0,0 +1,1069 @@
+### A Pluto.jl notebook ###
+# v0.19.25
+
+using Markdown
+using InteractiveUtils
+
+# This Pluto notebook uses @bind for interactivity. When running this notebook outside of Pluto, the following 'mock version' of @bind gives bound variables a default value (instead of an error).
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 746ff659-88ab-4ff4-8cba-43c798cacd3e
+using Images,TestImages, PlutoUI
+
+# ╔═╡ 04a5c6d4-f8d5-11ed-141a-35481b811ee9
+md"""
+## Images as Lists of numbers
+Hi There! Remember when you were kid and you used to play with legos ? you probably would put tiny pieces of different colors to form different shapes and elements ? 
+
+Well images in computers work exactly the same way! Each image is made of tiny elements we call pixels. Since our computers only understand numbers, these pixels are given to the computer as a list of numbers.  
+
+To keep things simple, we will only deal with black and white images for now. So, for a black and white image, each pixel is a number between 0 and 1. 
+
+Try it out! Move the slider around to get a different shade of gray
+"""
+
+# ╔═╡ c4da04ab-f7e0-46fd-b352-e518e4733608
+g_slider = @bind g PlutoUI.Slider(0:0.05:1)
+
+# ╔═╡ 96a4b35a-5a3a-4ad0-9ffe-306db46d1c03
+#Move the Slider: $(g_slider) $(pixel)
+pixel = Gray.(g)
+
+# ╔═╡ eb983abe-5899-431b-84a1-7d06715d75c4
+md"""
+Now that you know how images are built. Let's get to the fun part. First let's start by getting a test image
+"""
+
+# ╔═╡ ef0bea69-6dfd-4526-a96e-6607771660fe
+testimage("coffee")
+
+# ╔═╡ 04bdc84a-0e3b-48d7-9129-50bc8c421e7c
+md"""Pretty neat right? But we're sticking with black and white images. """
+
+# ╔═╡ ebb63415-2738-47f3-b0ef-0fd6e6bac259
+md"""
+## Using Filter 
+
+Now to make things more interesting, we will add some code that applies a filter to this image."""
+
+# ╔═╡ fcf738bd-ab29-4ad8-a837-0fb2d0e7ed85
+test_img = Gray.(testimage("coffee"))
+
+# ╔═╡ 44d594ff-a25b-4455-be56-863345c67b68
+# Code from "Towards Data Science": https://towardsdatascience.com/understanding-convolution-by-implementing-in-julia-3ed744e2e933
+function conv(input, filter)
+    input_r, input_c = size(input)
+    filter_r, filter_c = size(filter)
+
+    result = zeros(input_r-filter_r+1, input_c-filter_c+1)
+    result_r, result_c = size(result)
+
+    for i in 1:result_r
+        for j in 1:result_c
+            result[i,j] = sum(input[i:i+filter_r-1,j:j+filter_c-1].*filter)
+        end
+    end
+  
+    return result
+end
+
+# ╔═╡ 065819eb-3697-417d-9251-d7537cd3d247
+md"""Can you guess what this filter does?"""
```

Comment:
I like this type of interactivity, but without being explained what filtering is or without being invited to understand the `conv` function, this will be quite hard, no?